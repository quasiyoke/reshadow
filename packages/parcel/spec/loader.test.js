import compiler from './compiler.js';

describe('parcel', () => {
    it('inserts name and outputs JavaScript', async () => {
        const entry = './App/index.js';

        const {stats} = await compiler(entry);
        const output = stats.toJson();

        if (output.errors.length) {
            output.errors.forEach(console.error);

            throw new Error('there are some errors in compilation');
        }

        const appModule = output.modules.find(x => x.name === entry);

        expect(appModule.source).toMatchSnapshot();
        expect(
            stats.compilation.assets['styles.css'].source().toString(),
        ).toMatchSnapshot();
    });
});
