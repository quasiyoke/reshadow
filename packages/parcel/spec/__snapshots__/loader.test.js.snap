// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webpack inserts name and outputs JavaScript 1`] = `
"import React from 'react';
import styled, { set, create, css, use, map, __css__ } from \\"@reshadow/core\\";
const styled_c8 = create([(require('/.cache/reshadow/1409998408_1.css'), {
  \\"__button\\": \`___button_1m9qx_1\`,
  \\"_disabled\\": \`__disabled_1m9qx_1\`
})]);

const App = ({
  disabled,
  type,
  color = 'grey'
}) => styled((set([styled_c8], {
  \\"--c8_0\\": color
}), React.createElement(\\"button\\", map(\\"button\\", {
  type: type,
  disabled: disabled,
  $$style: styled.$$style
}, use({
  theme: \\"normal\\"
})), React.createElement(\\"div\\", {
  className: styled.styles[\\"__control\\"]
}, \\"click\\"))));

export default App;"
`;

exports[`webpack inserts name and outputs JavaScript 2`] = `
".___button_1m9qx_1.__disabled_1m9qx_1 {
    color: var(--c8_0);
    content: \\"'\\";
}"
`;
